src/Core/System.php

```diff
@@ -21,15 +21,206 @@
 
 namespace Friendica\Core;
 
+use Friendica\App;
+use Friendica\Core\Config\Capability\IManageConfigValues;
 use Friendica\DI;
 use Friendica\Network\HTTPException\InternalServerErrorException;
 use Friendica\Util\XML;
+use Psr\Log\LoggerInterface;
 
 /**
  * Contains the class with system relevant stuff
  */
 class System
 {
+	/**
+	 * @var LoggerInterface
+	 */
+	private $logger;
+
+	/**
+	 * @var App\Mode
+	 */
+	private $mode;
+
+	/**
+	 * @var IManageConfigValues
+	 */
+	private $config;
+
+	/**
+	 * @var string
+	 */
+	private $basePath;
+
+	public function __construct(LoggerInterface $logger, App\Mode $mode, IManageConfigValues $config, string $basepath)
+	{
+		$this->logger   = $logger;
+		$this->mode     = $mode;
+		$this->config   = $config;
+		$this->basePath = $basepath;
+	}
+
+	/**
+	 * Checks if the maximum number of database processes is reached
+	 *
+	 * @return bool Is the limit reached?
+	 */
+	public function isMaxProcessesReached(): bool
+	{
+		// Deactivated, needs more investigating if this check really makes sense
+		return false;
+
+		/*
+		 * Commented out to suppress static analyzer issues
+		 *
+		if ($this->mode->isBackend()) {
+			$process = 'backend';
+			$max_processes = $this->config->get('system', 'max_processes_backend');
+			if (intval($max_processes) == 0) {
+				$max_processes = 5;
+			}
+		} else {
+			$process = 'frontend';
+			$max_processes = $this->config->get('system', 'max_processes_frontend');
+			if (intval($max_processes) == 0) {
+				$max_processes = 20;
+			}
+		}
+
+		$processlist = DBA::processlist();
+		if ($processlist['list'] != '') {
+			$this->logger->debug('Processcheck: Processes: ' . $processlist['amount'] . ' - Processlist: ' . $processlist['list']);
+
+			if ($processlist['amount'] > $max_processes) {
+				$this->logger->debug('Processcheck: Maximum number of processes for ' . $process . ' tasks (' . $max_processes . ') reached.');
+				return true;
+			}
+		}
+		return false;
+		 */
+	}
+
+	/**
+	 * Checks if the minimal memory is reached
+	 *
+	 * @return bool Is the memory limit reached?
+	 */
+	public function isMinMemoryReached(): bool
```

Comment:
On Linux systems, this check doesn't make much sense since Linux will use as much memory as possible for internal caching, even when performance is nominal.

I think it should be deactivated like the `isMaxProcessesReached` test. Friendica has become a fully-fledged web application that requires external monitoring rather than these hacks.

Reply:
Done, see https://github.com/friendica/friendica/pull/10941/commits/b61532f77b02128a22ef9f850c6793b008424cb6